src/pkg/zoci/pull.go

```diff
@@ -74,85 +50,159 @@ func (r *Remote) PullPackage(ctx context.Context, destinationDir string, concurr
 	return layersToPull, nil
 }
 
-// LayersFromRequestedComponents returns the descriptors for the given components from the root manifest.
-//
-// It also retrieves the descriptors for all image layers that are required by the components.
-func (r *Remote) LayersFromRequestedComponents(ctx context.Context, requestedComponents []v1alpha1.ZarfComponent) ([]ocispec.Descriptor, error) {
-	layers := make([]ocispec.Descriptor, 0)
+// AssembleLayers returns all layers for the given zarf package to pull from OCI.
+func (r *Remote) AssembleLayers(ctx context.Context, requestedComponents []v1alpha1.ZarfComponent, isSkeleton bool, inspectTarget string) ([]ocispec.Descriptor, error) {
```

Comment:
Won't block on it, but it is worth considering adding unit tests to to this function.